src/relay/charpane.ash

```diff
@@ -4412,26 +4410,234 @@ void bakeBricks() {
+				//if a glue is found at the current portion of the string
+					loop = true;
+					if(length(ctoken) != 0){
+					//add non-zero length non-glue to list of tokens.
+						tokens[j] = ctoken;
+						set_length(ctoken,0);
+						j++;
+					}
+					//add found glue to list of tokens, and delete it from the front of str
+					tokens[j] = s;
+					str = substring(str,length(s));
+					j++;
+				}
+			}
+		}
+		if(length(str) > 0) {
+			//if no glue was found at the start of str, remove the first character from str
+			//and add it to ctoken. It will be added to the list of tokens when another glue
+			//or end-of-string is found.
+			append(ctoken, substring(str,0,1));
+			str = substring(str,1);
+			loop = true;
+		}
+	}
+	if(length(ctoken) > 0)
+		//add non-zero length last non-glue token to list of tokens.
+		tokens[j] = ctoken;
+	return tokens;
+}
+
+
+
+float starCount(string[int] styleInfos) {
+	//counts the number of stars "*" in the current row of bricks
+	//used to decide what css width should be set to a brick or group of bricks.
+	float starCountHelper(string styleInfo) {
+		buffer bnum;
+		string[int] tokens = tokenize(styleInfo, $strings[*,%,px]);
+		float num = 1;
+		foreach i,s in tokens {
+			switch(s){
+			// a number before a "*" will return that number. Everything else will return 1.
+				case "%":
+				case "px":
+					num = 1;
+					break;
+				case "*":
+					return num;
 				default:
-					result.append(chitBricks[brick]);
+					num = to_float(s);
+			}
+		}
+		return 1;
+	}
+	
+	float total = 0;
+	foreach i,s in styleInfos {
+		total += starCountHelper(s);
+	}
+	return total;
+}
+
+string stylize(string styleInfo, int total_stars) {
+//returns an appropriate width style to a brick or group of bricks, based on user's chit layout preference
+	string[int] tokens = tokenize(styleInfo, $strings[*,%,px]);
+	string style = "";
+	float num = 0;
+	foreach i,s in tokens {
+		switch(s){
+			case "*":
+				style += "width:" + (100.0 * max(num,1)/total_stars) +"%;";
+				break;
+			case "px":
+				style += "min-width:" + num + "px;";
+				style += "max-width:" + num + "px;";
+				break;
+			case "%":
+				style += "width:" + num + "%;";
+				break;
+			default:
+				num = to_float(s);
+		}
+	}
+	if(style == "")
+		style = "width:" + (100.0 * max(num,1)/total_stars) +"%;";
+	
+	return 'style="' + style + '"';
+}
+
+buffer addGroup(string[int] rowHTMLs, string className) {
+//adds rows of bricks into a vertical group
+	buffer buff;
+	if(className != "")
+		append(buff, '<div class='+className+'>');
+		
+	foreach i,s in rowHTMLs {
+		append(buff, s);
+	}
+	
+	if(className != "")
+		append(buff, '</div>');
+
+	return buff;
+}
+
+buffer addRow(string[int] bricks, string[int] styleInfos) {
+//adds bricks or groups of bricks into horizontal rows
+	buffer buff;
+	string brick;
+	int totalStars = starCount(styleInfos) + count(bricks) - count(styleInfos);
+	//append(buff, '<div class="brick_row">');
```

Comment:
If this is no longer necessary, please remove it (or is it just something you occasionally uncommented to help test with something for some reason? If that's the case, please add another comment explaining the use case there).

Reply:
It's no longer necessary. Removing it.